### PR TITLE
Initial commit for Communities feature

### PR DESCRIPTION
**Pull request by markwooff**

### CODE REVIEW

File: mod/gc_communities/start.php [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+            $ministries = json_decode($mins->ministries_fr, true);
+        }
+
+        $userObj = get_user_by_username($user);
+
+        if( $userObj ){
+
+            $userType = $userObj->user_type;
+            // if user is public servant
+            if( $userType == 'federal' ){
```
Comment (nschonni) [inline mod/gc_communities/start.php]: A `switch` might actually be easier to read

File: mod/gc_communities/views/default/ajax/community_feed.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+$limit = get_input('limit');
+$latest = get_input('latest');
+
+$url = get_input('url');
+foreach( $communities as $community ){
+    if( $community['community_url'] == $url ){
+        $community_en = $community['community_en'];
+        $community_fr = $community['community_fr'];
+        $community_tags = $community['community_tags'];
+        $community_animator = $community['community_animator'];
```
Comment (nschonni) [inline mod/gc_communities/views/default/ajax/community_feed.php]: `break;` after this so it doesn't keep looking once it's found the community?

File: mod/gc_communities/views/default/js/stream_community_feed.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+* @author Ilia github.com/phanoix
+* @author Mark github.com/markwooff
+**/
+
+$(document).ready(function(){
+    community_feed_stream_count(); 
+});
+
+// This function counts 2 minutes while a streaming page is loaded
+var interval = null;
```
Comment (nschonni) [inline mod/gc_communities/views/default/js/stream_community_feed.js]: You're adding this to the global/window scope. Might want to wrap this all in an IIFE

File: mod/gc_communities/views/default/tags/form.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    $community_tags = "";
+    foreach( $communities as $community ){
+        if( $community['community_url'] == $community_url ){
+            $community_tags = $community['community_tags'];
+        }
+    }
+?>
+
+<style type="text/css">
```
Comment (nschonni) [inline mod/gc_communities/views/default/tags/form.php]: Scoped styles probably won't work on many browsers https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style#Styles_scoped_to_the_parent_element_and_its_children

File: mod/gc_communities/views/default/widgets/filtered_feed_index/content.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Community page widgets
+ */
+ 
+ 	$widget = $vars['entity'];
+	// $object_types = array('object','group');
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_feed_index/content.php]: Delete

File: mod/gc_communities/views/default/widgets/filtered_feed_index/content.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Community page widgets
+ */
+ 
+ 	$widget = $vars['entity'];
+	// $object_types = array('object','group');
+	$object_types = array('object');
+	// $object_subtypes = array('blog','groupforumtopic','event_calendar','thewire');
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_feed_index/content.php]: Delete

File: mod/gc_communities/views/default/widgets/filtered_feed_index/content.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+	$object_subtypes = array('blog','groupforumtopic','event_calendar','file');
+
+	$widget->title = ( get_current_language() == "fr" ) ? $widget->widget_title_fr : $widget->widget_title_en;
+	
+	$num_items = $widget->num_items;
+	if ( !isset($num_items) ) $num_items = 10;
+
+	// $widget_groups = $widget->widget_groups;
+	// if ( !isset($widget_groups) ) $widget_groups = ELGG_ENTITIES_ANY_VALUE;
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_feed_index/content.php]: Delete?

File: mod/gc_communities/views/default/widgets/filtered_feed_index/content.php [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	    		$options['metadata_name_value_pairs'][] = array('name' => 'tags', 'value' => $tag, 'operand' => '=');
+	    		$options['metadata_name_value_pairs_operator'] = 'AND';
+			}
+		} else {
+			$options['metadata_name'] = 'tags';
+	    	$options['metadata_values'] = $widget_tags;
+		}
+	}
+
+	// if( !empty($widget_groups) && $widget_groups[0] != 0 ){
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_feed_index/content.php]: Delete?

File: mod/gc_communities/views/default/widgets/filtered_feed_index/content.php [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+	// $widget_groups = $widget->widget_groups;
+	// if ( !isset($widget_groups) ) $widget_groups = ELGG_ENTITIES_ANY_VALUE;
+
+	$widget_tags = trim($widget->widget_tags);
+  	if( $widget_tags ) $widget_tags = explode(',', $widget_tags);
+	
+	$widget_tag_logic = $widget->widget_tag_logic;
+
+	$options = array(
+		// 'type' => 'object',
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_feed_index/content.php]: Delete

File: mod/gc_communities/views/default/widgets/filtered_feed_index/content.php [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+		'types' => $object_types,
+		// 'subtype' => $object_type,
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_feed_index/content.php]: Delete

File: mod/gc_communities/views/default/widgets/filtered_members_index/content.php [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+		foreach($widget_users as $widget_user){
+			$userObj = get_user_by_username($widget_user);
+
+			if( $userObj ){
+
+				$userType = $userObj->user_type;
+			    // if user is public servant
+			    if( $userType == 'federal' ){
```
Comment (nschonni) [inline mod/gc_communities/views/default/widgets/filtered_members_index/content.php]: Switch/case?

**Comment by ToferC** [comment]
@markwooff - I think this is done. Pls. close if so.

**Comment by markwooff** [comment]
@nschonni I've made some tweaks to the code as per your suggestions. @Phanoix feel free to close PR after bringing in these changes.

**Pull request merged**